lib/spack/spack/package.py

```diff
@@ -983,28 +977,42 @@ def module(self):
                           fromlist=[self.__class__.__name__])
 
 
-    def setup_dependent_environment(self, module, spec, dependent_spec):
-        """Called before the install() method of dependents.
+    def environment_modifications(self, dependent_spec):
+        """
+        Called before the install() method of dependents.
+
+        Return the list of environment modifications needed by dependents (or extensions). Default implementation does
+        nothing, but this can be overridden by an extendable package to set up the install environment for its
+        extensions. This is useful if there are some common steps to installing all extensions for a certain package.
 
-        Default implementation does nothing, but this can be
-        overridden by an extendable package to set up the install
-        environment for its extensions.  This is useful if there are
-        some common steps to installing all extensions for a
-        certain package.
+        Example :
 
-        Some examples:
+        1. Installing python modules generally requires `PYTHONPATH` to point to the lib/pythonX.Y/site-packages
+        directory in the module's install prefix.  This could set that variable.
 
-        1. Installing python modules generally requires PYTHONPATH to
-           point to the lib/pythonX.Y/site-packages directory in the
-           module's install prefix.  This could set that variable.
+        2. A lot of Qt extensions need `QTDIR` set.  This can be used to do that.
 
-        2. Extensions often need to invoke the 'python' interpreter
-           from the Python installation being extended.  This routine can
-           put a 'python' Execuable object in the module scope for the
-           extension package to simplify extension installs.
+        Args:
+            dependent_spec: dependent (or extension) of this spec
```

Comment:
@tgamblin It could make sense for things like MPI libraries, that have to set `MPI_CC` like variables to the compiler wrapper when installing and *may* want to set them to the real compiler in modules